controlplane/kubeadm/controllers/controller_test.go

```diff
@@ -158,21 +158,27 @@ func TestReconcileUpdateObservedGeneration(t *testing.T) {
 	g.Expect(testEnv.Create(ctx, kcp)).To(Succeed())
 
 	// read kcp.Generation after create
-	errGettingObject := testEnv.Get(ctx, util.ObjectKey(kcp), kcp)
-	g.Expect(errGettingObject).NotTo(HaveOccurred())
+	g.Eventually(func() error {
```

Comment:
I don't really know the code around here. But does the eventually change anything? The create line l.158 should be synchronous and the first get should return here. It doesn't really wait until a reconcile is done.

(sorry if this doesn't make sense)

Reply:
Get uses a cache internally, and the cache gets aligned asynchronously by the informers, and a a consequence there is a chance of stale reads in case of resource starvation...